root/entity/Details.js

```diff
@@ -49,8 +49,12 @@ const WSLink = ({
   entityType === 'release' && inc.push('labels', 'discids', 'recordings');
   const protocol = isSecureConnection ? 'https://' : 'http://';
   const link = '/ws/2/' + entityTypeForUrl + '/' + entityGid +
-               (inc.length ? '?inc=' : '') + inc.join('+') +
-               (isJson ? '&fmt=json' : '');
+               (inc.length ? '?inc=' + inc.join('+') : '') +
+               (isJson && inc.length
+                 ? '&fmt=json'
+                 : isJson
+                   ? '?fmt=json'
+                   : '');
```

Comment:
to remove scary double-ternary operator, what about making an array of query params and joining it?
```js
const params = []
if (inc.length) {
  params.push('inc=' + inc.join('+'))
}
if (isJson) {
  params.push('fmt=json')
}
if (params.length) {
  // or do a ternary when making const.link as you currently do?
  link = link + '?' + params.join('&')
}
```

a bit longer, but more explicit as to what's happening. I'd have suggested [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) but it's still unsupported in a few older browsers

Reply:
This component only runs on the server, so we could use `URLSearchParams` as it's supported by Node.

Reply:
Fine, converted this to use URL in general.